### PR TITLE
Fix guidance usage in continuous decoding

### PR DESCRIPTION
**Pull request by kunal-vaishnavi**
### Description

This PR resets the guidance logits processor when generation is completed and fixes the last action that occurred.

### Motivation and Context

This PR fixes a bug with providing grammars in multi-turn conversations where the generator is re-used.

### CODE REVIEW

**Review by JC1DA — APPROVED** [review]
LGTM
